BUILD

```diff
@@ -0,0 +1,522 @@
+        "c++/src/capnp/rpc.h",
+        "c++/src/capnp/rpc-prelude.h",
+        "c++/src/capnp/rpc-twoparty.capnp.h",
+        "c++/src/capnp/rpc-twoparty.h",
+    ],
+    copts = [
+        "-Wno-shadow",
+        "-Wno-unused-parameter",
+    ],
+    include_prefix = "capnp",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":capnp",
+        "//:kj",
+        "//:kj-async",
+    ],
+)
+
+cc_import(
+    name = "capnp-json_schemas",
+    hdrs = [
+        "c++/src/capnp/compat/json.h",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "capnp-json",
+    srcs = [
+        "c++/src/capnp/compat/json.c++",
+        "c++/src/capnp/compat/json.capnp.c++",
+    ],
+    hdrs = [
+        "c++/src/capnp/compat/json.capnp.h",
+        "c++/src/capnp/compat/json.h",
+    ],
+    copts = [
+        "-Wno-shadow",
+        "-Wno-unused-parameter",
+    ],
+    include_prefix = "capnp",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":capnp",
+        "//:kj",
+        "//:kj-async",
+    ],
+)
+
+cc_library(
+    name = "capnp-websocket",
+    srcs = [
+        "c++/src/capnp/compat/websocket-rpc.c++",
+    ],
+    hdrs = [
+        "c++/src/capnp/compat/websocket-rpc.h",
+    ],
+    copts = [
+        "-Wno-shadow",
+        "-Wno-unused-parameter",
+    ],
+    include_prefix = "capnp",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":capnp",
+        ":capnp-rpc",
+        "//:kj",
+        "//:kj-async",
+        "//:kj-http",
+    ],
+)
+
+cc_library(
+    name = "capnpc",
+    srcs = [
+        "c++/src/capnp/compiler/compiler.c++",
+        "c++/src/capnp/compiler/compiler.h",
+        "c++/src/capnp/compiler/error-reporter.c++",
+        "c++/src/capnp/compiler/error-reporter.h",
+        "c++/src/capnp/compiler/generics.c++",
+        "c++/src/capnp/compiler/generics.h",
+        "c++/src/capnp/compiler/grammar.capnp.c++",
+        "c++/src/capnp/compiler/lexer.c++",
+        "c++/src/capnp/compiler/lexer.capnp.c++",
+        "c++/src/capnp/compiler/lexer.capnp.h",
+        "c++/src/capnp/compiler/node-translator.c++",
+        "c++/src/capnp/compiler/node-translator.h",
+        "c++/src/capnp/compiler/parser.c++",
+        "c++/src/capnp/compiler/resolver.h",
+        "c++/src/capnp/compiler/type-id.c++",
+        "c++/src/capnp/compiler/type-id.h",
+        "c++/src/capnp/schema-parser.c++",
+        "c++/src/capnp/serialize-text.c++",
+    ],
+    hdrs = [
+        "c++/src/capnp/compiler/compiler.h",
+        "c++/src/capnp/compiler/grammar.capnp.h",
+        "c++/src/capnp/compiler/lexer.capnp.h",
+        "c++/src/capnp/compiler/lexer.h",
+        "c++/src/capnp/compiler/parser.h",
+    ],
+    copts = [
+        "-Wno-shadow",
+        "-Wno-unused-parameter",
+    ],
+    include_prefix = "capnp",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":capnp",
+        "//:kj",
+    ],
+)
+
+cc_binary(
+    name = "capnp_tool",
```

Comment:
`bin` or `capnp_bin` is probably a better name. I didn't realise what this was at first because of `tool`.